src/masternodes/rpc_evm.cpp

```diff
@@ -122,11 +139,186 @@ UniValue evmtx(const JSONRPCRequest& request) {
+        case EvmMapType::EVM_ADDRESS_TO_DVM: {
+            const CPubKey key = AddrToPubKey(pwallet, hash);
+            return EncodeDestination(PKHash(key.GetID()));
+        }
+        case EvmMapType::DVM_TX_TO_EVM: {
+            const auto res = pcustomcsview->GetTxHash(CEvmDvmMapType::DvmEvm, uint256S(hash));
+            if (!res) {
+                throw JSONRPCError(RPC_INVALID_REQUEST, res.msg);
+            } else {
+                return res.val->ToString();
+            }
+        }
+        case EvmMapType::EVM_TX_TO_DVM: {
+            const auto res = pcustomcsview->GetTxHash(CEvmDvmMapType::EvmDvm, uint256S(hash));
+            if (!res) {
+                throw JSONRPCError(RPC_INVALID_REQUEST, res.msg);
+            } else {
+                return res.val->ToString();
+            }
+        }
+        case EvmMapType::DVM_BLOCK_TO_EVM: {
+            const auto res = pcustomcsview->GetBlockHash(CEvmDvmMapType::DvmEvm, uint256S(hash));
+            if (!res) {
+                throw JSONRPCError(RPC_INVALID_REQUEST, res.msg);
+            } else {
+                return res.val->ToString();
+            }
+        }
+        case EvmMapType::EVM_BLOCK_TO_DVM: {
+            const auto res = pcustomcsview->GetBlockHash(CEvmDvmMapType::EvmDvm, uint256S(hash));
+            if (!res) {
+                throw JSONRPCError(RPC_INVALID_REQUEST, res.msg);
+            } else {
+                return res.val->ToString();
+            }
+        }
+        case EvmMapType::AUTO: {
+            return "";
+        }
+        default:
+            break;
+    }
+}
+
+UniValue logxvmindexes(const JSONRPCRequest& request) {
+    RPCHelpMan{
+            "logxvmindexes",
+            "\nLogs all block or tx indexes for debugging.\n",
+            {
+                    {"type", RPCArg::Type::NUM, RPCArg::Optional::NO, "Type of log: 0 - Blocks, 1 - Txs"}
+                },
+            RPCResult{
+                    "{...} (array) Json object with account balances if rpcresult is enabled."
+                    "This is for debugging purposes only.\n"},
+            RPCExamples{
+                    HelpExampleCli("logxvmindexes", R"('"<hex>"' 1)")},
+    }.Check(request);
+
+    LOCK(cs_main);
+
+    size_t count{};
+    UniValue result{UniValue::VOBJ};
+    UniValue indexesJson{UniValue::VOBJ};
+    const auto type = static_cast<EvmLogType>(request.params[0].get_int());
+    switch (type) {
+        case EvmLogType::BLOCK: {
+            pcustomcsview->ForEachBlockIndexes([&](const std::pair<uint8_t, uint256> &index, uint256 blockHash) {
+                if (index.first == CEvmDvmMapType::DvmEvm) {
+                    indexesJson.pushKV(index.second.GetHex(), blockHash.GetHex());
+                    ++count;
+                }
+                return true;
+            });
+        }
+        case EvmLogType::TX: {
+            pcustomcsview->ForEachTxIndexes([&](const std::pair<uint8_t, uint256> &index, uint256 txHash) {
```

Comment:
Lock cs_main before calling pcustomcsview.

Reply:
I'm already locking it at the beginning of the function